### DISCUSSION TITLE
Recommended way to override primary color per tenant basis

### DISCUSSION THREAD

**Discussion by sakhmedbayev** [tailwindlabs/tailwindcss#3100] [General]
We are building a multiple-tenant app using Next.js and Tailwindcss, where each tenant can adjust his/her app's theme, so they can change, for instance, a primary color. This color will be saved in the tenant's config. 

Question: what would be a recommended way to accomplish something like this using tailwind? 

We understand that we can make `primary-color` to work as a js variable, put it somewhere in the app's state, and then derive it from there, however, we wonder if there is a more efficient way to do this.

**Comment by mainej** [comment]
You could adapt the ideas in #1978, that is, use CSS variables. The technique boils down to doing 3 things:
1. In `tailwind.config.js`, define a `primary` color which references an (undefined) CSS variable.
2. Define utility classes that set the CSS variable, e.g `.primary-orange` and `.primary-green`
3. Use JS to put the tenant's preferred color class on the `<html>` or `<body>` tag, e.g. `<html class="primary-orange">`

Then, you would use `class="bg-primary"` or `class="border-primary"` on descendant tags. Because of the cascade, the CSS variable will be defined for all child tags.

This technique gets a little trickier if you want to support many hues of the primary color, or support transparency, or have both primary and secondary colors.
